README.md

```diff
@@ -11,44 +11,104 @@
+   This currently cannot be controlled nor configured by the user.
+2. A per-rule `matchAtMost` counter (`Process this rule at most (times)` in the UI) that limits the number of times a rule can be applied
+   (default is `-1`, meaning there is no limit).
+3. A per-rule `continueAfterMatch` boolean (`Proceed to the next rule if the conditions of this rule are met` in the UI)
+   that causes the plugin to apply the next rule to the same build after the current rule has been applied
+   (default is `true`, meaning to continue to apply rules to the build being processed).
+
+### Optional Conditions
+
+Users can add a list of [Condition](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/Condition.java)s to each rule,
+in any order they need.
+The conditions are checked in the order they are defined.
+
+The conditions are:
+
+- [Build age range](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/BuildAgeRangeCondition.java)
+- [Build cause](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/CauseCondition.java)
+- [Build number range](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/BuildNumberRangeCondition.java)
+- [Build result](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/BuildResultCondition.java)
+- [Match every build](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/MatchEveryBuildCondition.java)
+- [Token Macro](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/TokenMacroCondition.java)
+
+Providing the built-in conditions allow, the optional condition checks have the following effects:
+
+1. As soon as one of the conditions does not match, the plugin ignores the rule's Actions.
+2. When all the conditions of the rule are met, the plugin applies the rule's Actions.
+3. When no optional conditions are specified, the plugin automatically applies the rule's Actions.
+
+### Optional Actions
+
+Users can add a list of [Action](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/Action.java)s to each rule,
+in any order they need.
+The actions are applied in the order they are defined.
+
+The actions are:
+
+- [Change build description](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/ChangeBuildDescriptionAction.java)
+- [Delete artifacts](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsAction.java)
+- [Delete build](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteBuildAction.java)
+- [Delete log file](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteLogFileAction.java) 
+
+If the first action is to delete the build, the other actions will still be applied but have no real effect.
```

Comment:
I'm not so sure.
Build deletion is not performed synchronously so I can imagine configuration and runtime execution when first action is to delete the build and second is to keep the build forever. I'm not sure what happens then :-)